### DISCUSSION TITLE
permissions error for originals folder

### DISCUSSION THREAD

**Discussion by scblakely** [photoprism/photoprism#4247] [Q&A] [answered]
I have Photoprism running in an LXC on Proxmox. I have mounted my photos via CIFS from my NAS, and passed the folder into the LXC as a bind mount. This works correctly, and root within the LXC has write permissions to the volume.

I have set that folder as my originals folder in Photoprism.
```
PHOTOPRISM_AUTH_MODE='password'
PHOTOPRISM_ADMIN_PASSWORD='xxxxxxxxx'
PHOTOPRISM_HTTP_HOST='0.0.0.0'
PHOTOPRISM_HTTP_PORT='2342'
PHOTOPRISM_SITE_CAPTION='https://tteck.github.io/Proxmox/'
PHOTOPRISM_STORAGE_PATH='/opt/photoprism/storage'
PHOTOPRISM_ORIGINALS_PATH='/home1/photos'
PHOTOPRISM_IMPORT_PATH='/home1/photo_import'
```
```
root@photoprism:/opt/photoprism/bin# ls -la /home1/
total 9
drwxr-xr-x  5 root root  5 May 10 20:20 .
drwxr-xr-x 18 root root 22 May 10 20:22 ..
drwxr-xr-x  2 root root  0 May 10 17:24 photo_import
drwxr-xr-x  2 root root  0 May  9 18:10 photos
drwxr-xr-x  2 root root  0 May 10 20:30 storage
```
So far, so good. The only thing I could not do was get it to work with the storage folder set to /home1/storage on the NAS (I am guessing that SQLite takes exception to the CIFS mount). Problem for another day - I'll probably use a mariadb SQL server.

After an error tagging faces (I accidently merged two individuals), I decided to reset the face tagging with **photoprism faces reset -f** but this does not work
```
root@photoprism:/opt/photoprism/bin# ./photoprism faces reset -f
Permanently remove all people and faces?: Y
WARN[2024-05-11T16:27:12+12:00] config: invalid originals path, check configuration and permissions when loading plus extension
ERRO[2024-05-11T16:27:12+12:00] config: invalid originals path, check configuration and permissions
```
What causes this error, and how can I fix it?

**Comment by lastzero** [comment]
Please use the "photoprism config" command to see a list of all config options and their current values to check if they match your settings in this... ".env" file? Note that quotes ' might not be supported there, so one idea would be to try without. Also make sure the configured paths are all readable and writable, as indicated by the error message you see.

**Reply by scblakely** [reply]
Yeah - the output of **photoprism config** isn't reflecting the values I expect, and _originals-path_ is blank.

I'll have another go to see if I set the options another way.

**Reply by lastzero** [reply]
Note that you can alternatively use config files as described here in our documentation:
- https://docs.photoprism.app/getting-started/config-files/

**Reply by scblakely** [reply, marked as answer]
Yep - created the config files in **/etc/photoprism** and everything now seems to be running as I expected.

Now to get the MariaDB database set up and the storage on to a bind mount as well (for easy backup).

Cheers.

**Reply by csrgzz** [reply]
> Yep - created the config files in **/etc/photoprism** and everything now seems to be running as I expected.
> 
> Now to get the MariaDB database set up and the storage on to a bind mount as well (for easy backup).
> 
> Cheers.
@scblakely 
Did you by any chance modified the .env file on /opt/photoprism/config and added PHOTOPRISM_CONFIG_PATH=/etc/photoprism/defaults.yml and then created and added all your configs here  /etc/photoprism/defaults.yml and reload?
I'm getting lots of database errors when scanning so wonder if I should try this to fix those issues as well as the issue with invalid original path...

Thanks

**Reply by hacs64** [reply]
@scblakely Same question. Could you describe steps?

**Comment by hellofaduck** [comment]
I couldn't figure out the magic of configs from the console, so I came up with a workaround, I need to do this very rarely, so I control the users by just substituting the service startup string in /etc/systemd/system/photoprism.service
I think if you create the same variables under the current user as in the env file you can achieve success too, but that's fine with me.
